streams/src/test/java/org/apache/kafka/streams/processor/internals/StateConsumerTest.java

```diff
@@ -108,15 +108,6 @@ public void shouldNotFlushOffsetsWhenFlushIntervalHasNotLapsed() {
         assertFalse(stateMaintainer.flushed);
     }
 
-    @Test
-    public void shouldNotFlushWhenFlushIntervalIsZero() {
-        stateConsumer = new GlobalStreamThread.StateConsumer(logContext, consumer, stateMaintainer, time, Duration.ofMillis(10L), -1);
```

Comment:
From my understanding, the test name is wrong and should be like `shouldNotFlushWhenFlushIntervalIsNegative`.
The code that `shouldNotFlushWhenFlushIntervalIsZero` tests is about this line:

https://github.com/apache/kafka/blob/74f686d3c24d06c977341cee7294f36e02cbc366/streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java#L244

With the code on trunk, when `flushInterval` is set to negative, `stateMaintainer.flushState()` is not called.
But, since this spec is removed, I think it is okay to remove the test.